internal/addon/addon.go

```diff
@@ -30,3 +39,83 @@ func GetObjectKey(configRef []addonapiv1alpha1.ConfigReference, group, resource
 	}
 	return key
 }
+
+func AgentHealthProber() *agent.HealthProber {
+	return &agent.HealthProber{
+		Type: agent.HealthProberTypeDeploymentAvailability,
+		WorkProber: &agent.WorkHealthProber{
+			ProbeFields: []agent.ProbeField{
+				{
+					ResourceIdentifier: workapiv1.ResourceIdentifier{
+						Group:     OtelcolGroup,
+						Resource:  OtelcolResource,
+						Name:      OtelcolName,
+						Namespace: OtelcolNS,
+					},
+					ProbeRules: []workapiv1.FeedbackRule{
+						{
+							Type: workapiv1.JSONPathsType,
+							JsonPaths: []workapiv1.JsonPath{
+								{
+									Name: "replicas",
+									Path: ".spec.replicas",
+								},
+							},
+						},
+					},
+				},
+				{
+					ResourceIdentifier: workapiv1.ResourceIdentifier{
+						Group:     ClfGroup,
+						Resource:  ClfResource,
+						Name:      ClfName,
+						Namespace: ClusterLoggingNS,
+					},
+					ProbeRules: []workapiv1.FeedbackRule{
+						{
+							Type: workapiv1.JSONPathsType,
+							JsonPaths: []workapiv1.JsonPath{
+								{
+									Name: "type",
+									Path: ".status.conditions[0].type",
+								},
+							},
+						},
+					},
+				},
+			},
+			HealthCheck: func(identifier workapiv1.ResourceIdentifier, result workapiv1.StatusFeedbackResult) error {
+				for _, value := range result.Values {
+					if identifier.Resource == ClfResource {
```

Comment:
I think I would change the if... else if... else to a switch statement, that might make it more easy read